tools/lsp/preview/ui.rs

```diff
@@ -234,147 +236,129 @@ fn map_property_declaration(
-    let mut opening_braces = 0;
-
-    let mut result = String::new();
-
-    for c in value.chars() {
-        if last_was_quote || (!is_first && !had_initial_quote) {
-            return None;
-        }
+fn extract_tr_data(tr_node: &syntax_nodes::AtTr) -> Option<(String, String, String)> {
+    let text = tr_node
+        .child_text(SyntaxKind::StringLiteral)
+        .and_then(|s| i_slint_compiler::literals::unescape_string(&s))?;
+    let context = tr_node
+        .TrContext()
+        .and_then(|n| n.child_text(SyntaxKind::StringLiteral))
+        .and_then(|s| i_slint_compiler::literals::unescape_string(&s))
+        .unwrap_or_default();
+    let plural = tr_node
+        .TrPlural()
+        .and_then(|n| n.child_text(SyntaxKind::StringLiteral))
+        .and_then(|s| i_slint_compiler::literals::unescape_string(&s))
+        .unwrap_or_default();
+    Some((context, plural, text))
+}
 
-        match c {
-            '"' => {
-                if is_first {
-                    had_initial_quote = true;
-                } else if is_escaped {
-                    result.push(c);
-                    is_escaped = false;
-                } else {
-                    last_was_quote = true;
-                }
-            }
-            '\\' => {
-                result.push(c);
-                is_escaped = !is_escaped;
-            }
-            'n' => {
-                result.push(c);
-                is_escaped = false;
-            }
-            'u' => {
-                result.push(c);
-                if is_escaped {
-                    about_to_unicode_escape = true;
-                    is_escaped = false;
-                }
-            }
-            '{' => {
-                result.push(c);
-                if in_expression_escape {
-                    opening_braces += 1;
-                } else if about_to_unicode_escape {
-                    about_to_unicode_escape = false;
-                    in_unicode_escape = true;
-                } else if is_escaped {
-                    in_expression_escape = true;
-                    is_escaped = false;
+fn simplify_value(
+    document_cache: &common::DocumentCache,
+    element_node: &common::ElementRcNode,
+    property_type: &str,
+    code_block_or_expression: &Option<properties::CodeBlockOrExpression>,
+) -> PropertyValue {
+    let expression = code_block_or_expression.as_ref().and_then(|cbe| cbe.expression());
+
+    let mut kind = PropertyValueKind::Code;
+    let mut value_bool = false;
+    let mut value_int = 0;
+    let mut value_float = 0.0;
+    let mut value_string = String::new();
+    let mut visual_items = Rc::new(VecModel::default()).into();
+    let mut is_translatable = false;
+    let mut tr_context = String::new();
+    let mut tr_plural = String::new();
+    let mut default_selection = 0;
+    let code =
+        code_block_or_expression.as_ref().map(|cbe| cbe.text().to_string()).unwrap_or_default();
+
+    match property_type {
```

Comment:
Can you not match on the langtype::Type here?

Reply:
That is probably a good idea.